plotting/prepare_data.py

```diff
@@ -0,0 +1,91 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Prepares raw 'XYE' data to be plotted by converting it into a data frame.
+"""
+from difflib import ndiff
+
+import csv
+import pandas as pd
+
+
+class PrepareData:
+    # pylint:disable=too-few-public-methods
+    """
+    This class prepares data to be plotted by first reading it from a given path,
+    validating initial rows are as expected, and ultimately converting the data
+    into a pandas data frame.
+    """
+
+    def __init__(self):
+        self.expected_first_row = "# X , Y , E"
+        self.columns = ["Spectrum", "X", "Y", "E"]
+
+    def prepare_data(self, path):
+        """
+        Reads raw XYE data from a given path.
+        Validates initial 2 rows are as expected.
+        Converts the data into a pandas.DataFrame
+        :param path: The location of the XYE data to be prepared
+        :return: The pandas.DataFrame comprising the data
+        """
+        processed_data = []
+        with open(path, 'r') as input_file:
+            self._check_first_row(input_file.readline())
+            spectrum = self._check_second_row(input_file.readline())
+
+            reader = csv.reader(input_file)
+            for row in reader:
+                float_list = [float(item) for item in row]
+                if len(float_list) is 1:     # pylint:disable=literal-comparison
```

Comment:
Should use `len(float_list) == 1:`
`is` should be used for direct comparison by reference. see SO article for a pretty good explanation https://stackoverflow.com/questions/47778889/pylint-complains-about-comparing-a-string-to-a-literal-with-is
The example here is with strings, but the same applies

Reply:
ps. this is also why pylint is complaining